test/core/extensions/coreIssuance.spec.ts

```diff
@@ -475,4 +475,167 @@ contract("CoreIssuance", (accounts) => {
+
+    beforeEach(async () => {
+      components = await erc20Wrapper.deployTokensAsync(numComponents, ownerAccount);
+      await erc20Wrapper.approveTransfersAsync(components, transferProxy.address);
+
+      const componentAddresses = _.map(components, (token) => token.address);
+      componentUnits = _.map(components, () => naturalUnit.mul(2)); // Multiple of naturalUnit
+      setToken = await coreWrapper.createSetTokenAsync(
+        core,
+        setTokenFactory.address,
+        componentAddresses,
+        componentUnits,
+        naturalUnit,
+      );
+
+      await coreWrapper.issueSetTokenAsync(core, setToken.address, naturalUnit);
+
+      subjectCaller = ownerAccount;
+      subjectQuantityToRedeem = naturalUnit;
+      subjectSetToRedeem = setToken.address;
+      subjectComponentsToWithdrawMask = coreWrapper.maskForAllComponents(numComponents);
+    });
+
+    async function subject(): Promise<string> {
+      return core.redeemAndWithdraw.sendTransactionAsync(
+        subjectSetToRedeem,
+        subjectQuantityToRedeem,
+        subjectComponentsToWithdrawMask,
+        { from: ownerAccount },
+      );
+    }
+
+    it("decrements the balance of the tokens owned by set in vault", async () => {
+      const existingVaultBalances = await coreWrapper.getVaultBalancesForTokensForOwner(components, vault, subjectSetToRedeem);
+
+      await subject();
+
+      const expectedVaultBalances = _.map(components, (component, idx) => {
+        const requiredQuantityToRedeem = subjectQuantityToRedeem.div(naturalUnit).mul(componentUnits[idx]);
+        return existingVaultBalances[idx].sub(requiredQuantityToRedeem);
+      });
+      const newVaultBalances = await coreWrapper.getVaultBalancesForTokensForOwner(components, vault, subjectSetToRedeem);
+      expect(newVaultBalances).to.eql(expectedVaultBalances);
+    });
+
+    it("decrements the balance of the set tokens owned by owner", async () => {
+      const existingSetBalance = await setToken.balanceOf.callAsync(ownerAccount);
+
+      await subject();
+
+      const expectedSetBalance = existingSetBalance.sub(subjectQuantityToRedeem);
+      const newSetBalance = await setToken.balanceOf.callAsync(ownerAccount);
+      expect(newSetBalance).to.be.bignumber.equal(expectedSetBalance);
+    });
+
+    it("transfers all of the component tokens back to the user", async () => {
+      const existingTokenBalances = await erc20Wrapper.getTokenBalances(components, ownerAccount);
+
+      await subject();
+
+      const expectedNewBalances = _.map(existingTokenBalances, (balance, idx) => {
+        const quantityToRedeem = subjectQuantityToRedeem.div(naturalUnit).mul(componentUnits[idx]);
+        return balance.add(quantityToRedeem);
+      });
+      const newTokenBalances = await erc20Wrapper.getTokenBalances(components, ownerAccount);
+      expect(newTokenBalances).to.eql(expectedNewBalances);
+    });
+
+    describe("when the withdraw mask includes one component", async () => {
```

Comment:
Is it worth it to also include a check to make doubly sure that no component tokens have been transferred to the user that weren't supposed to be transferred, purely to prevent any potential regressions. I don't feel overly strongly...could be over testing.